src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py

```diff
@@ -239,17 +241,21 @@ def register_source_uri_arguments(scope):
 for item in ['blob', 'file', 'queue', 'table']:
     register_cli_argument('storage account show-connection-string', '{}_endpoint'.format(item), help='Custom endpoint for {}s.'.format(item))
 
+register_cli_argument('storage account create', 'account_type', help='The storage account type', **enum_choice_list(get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='AccountType')))
+
 register_cli_argument('storage account create', 'account_name', account_name_type, options_list=('--name', '-n'), completer=None)
 
-register_cli_argument('storage account create', 'kind', help='Indicates the type of storage account.', **enum_choice_list(Kind))
+register_cli_argument('storage account create', 'kind', help='Indicates the type of storage account.', **enum_choice_list(get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='Kind')))
 register_cli_argument('storage account create', 'tags', tags_type)
 
 for item in ['create', 'update']:
-    register_cli_argument('storage account {}'.format(item), 'sku', help='The storage account SKU.', **enum_choice_list(SkuName))
-    register_cli_argument('storage account {}'.format(item), 'encryption', nargs='+', help='Specifies which service(s) to encrypt.', validator=validate_encryption, **enum_choice_list(list(EncryptionServices._attribute_map.keys())))  # pylint: disable=protected-access
+    register_cli_argument('storage account {}'.format(item), 'sku', help='The storage account SKU.', **enum_choice_list(get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='SkuName')))
+    es_model = get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='EncryptionServices')
+    if es_model:
+        register_cli_argument('storage account {}'.format(item), 'encryption', nargs='+', help='Specifies which service(s) to encrypt.', validator=validate_encryption, **enum_choice_list(list(es_model._attribute_map.keys())))  # pylint: disable=protected-access
```

Comment:
In other cases, `enum_choice_list` handles it as it can deal with `None` values which is what `get_versioned_models` returns if the model doesn't exist.

This is a special case as it's not the model that we pass into `enum_choice_list`, it's `model._attribute_map.keys()` and so if `model` is `None`, I get an AttributeError on `_attribute_map`.
That's why in *this case*, I have an if statement before I register the argument.